notebook/04_portfolio_trade.ipynb

```diff
@@ -441,84 +423,99 @@
     }
    ],
    "source": [
-    "o_cpu = task_graph.run(\n",
-    "            outputs=['node_sharpeRatio', 'node_cumlativeReturn',\n",
-    "                     'node_csvdata'],\n",
-    "            replace={'node_csvdata': {\"type\": \"PandasCsvStockLoader\"},\n",
-    "                     'node_filterValue': {\"conf\": [{\"column\": \"volume_mean\", \"min\": min_volume},\n",
-    "                                                   {\"column\": \"returns_max\", \"max\": max_rate},\n",
-    "                                                   {\"column\": \"returns_min\", \"min\": min_rate}]},\n",
-    "                     'node_addReturn': {\"type\": \"CpuReturnFeatureNode\"},\n",
-    "                     'node_addIndicator': {\"type\": \"CpuAssetIndicatorNode\"},\n",
-    "                     'node_exp_strategy': {\"type\": \"CpuPortExpMovingAverageStrategyNode\"}})\n",
-    "cpu_input_cached = o_cpu[2]\n",
     "plot_figures(o_cpu)"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "It produces the exact same results as GPU version but much slower. All the above computation happens in the CPU Pandas Dataframe. To quantify the performance, we do the following experiment:"
+    "## Benchmarks\n",
+    "\n",
+    "While running this notebook, we have obtained the following results:\n",
+    "\n",
+    "- 73 seconds to run in CPU (Intel(R) Xeon(R) Gold 6148).\n",
+    "-  4 seconds to run in GPU (NVIDIA v100).\n",
+    "\n",
+    "We get 67x speed up by using GPU and GPU dataframes, compared to CPU and CPU dataframes.\n",
```

Comment:
67x speedup is wrong here. We need to reevaluate the performance as RAPIDS 0.8 is faster now.

Reply:
> Hi Yi,
> 
> I have completed everything but this one:
> 
> > we want to point out the user needs to set "self.delayed_process = True" to enable this dask computation graph integration.
> 
> Could you please write the complete sentence? I cannot see either where that comment should be.
> 
> Thanks!
if inside the processor method, customized GPU kernel functions are used to process the dataframe instead of the the normal datarame api functions calls, we need to add "self.delayed_process = True" in the column_setup method to let the gQuant handle the dask graph integration problem. 
If we only use normal dataframe api functions inside the process method, nothing need to be done as self.delayed_process = False by default.

Reply:
Done!